src/stash/src/postgres.rs

```diff
@@ -413,7 +423,7 @@ impl Metrics {
 /// migration path.
 pub struct Stash {
     txn_mode: TransactionMode,
-    url: String,
+    config: Arc<tokio::sync::Mutex<Config>>,
```

Comment:
Why is the mutex here needed? I thought that we always interact with the Stash on a single thread.